impl/src/main/java/org/jboss/cdi/tck/tests/beanContainer/BeanContainerTest.java

```diff
@@ -161,4 +168,46 @@ public void testGetContexts() {
         assertTrue(builtInContextImpls.size() >= 1);
     }
 
+    @Test
+    @SpecAssertion(section = BM_PROXY_UNWRAP, id = "a")
+    public void testUnwrapClientProxyForNormalScope() {
+        ApplicationScopedBean applicationScoped = getContextualReference(ApplicationScopedBean.class);
+        assertNotSame(applicationScoped, getCurrentBeanContainer().unwrapClientProxy(applicationScoped));
+    }
```

Comment:
In this test, could we call `getClass()` on the unwrapped proxy and assert that the class is `ApplicationScopedBean.class` and not a subclass?